### ISSUE TITLE
内积计算方式可能有问题？

### ISSUE THREAD

**Issue by xgxgnpu** [tianheyu927/PCGrad#15]
您好！谢谢您的开源代码，非常棒的工作，但我对代码实现有一些疑问。

1、在您的代码第46行，计算梯度向量内积似乎有点问题，当然也有可能是我没有完全理解您的算法，请您见谅！
`inner_product = tf.reduce_sum(grad_task*grads_task[k])`

这行代码是为了计算梯度向量的内积，比如说有2个task loss，那么grad_task是二维张量（里面是两个loss的梯度向量），而grads_task[k]是一维张量（里面是其中一个loss的梯度向量），这样内积再求和，得到的也不是两个向量的内积。请问您是怎么理解这一行代码的呢？谢谢！

**Comment by XiangqiUESTC** [comment]
grad_task应该是一个任务的梯度
